src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyBuilder.cs

```diff
@@ -36,6 +37,21 @@ public CorsPolicyBuilder(CorsPolicy policy)
         /// </summary>
         /// <param name="origins">The origins that are allowed.</param>
         /// <returns>The current policy builder.</returns>
+        /// <remarks>
+        /// This method normalizes the origin headers prior to adding it to <see cref="CorsPolicy.Origins"/> to match
+        /// the normalization performed by the browser on the value sent in the <c>ORIGIN</c> header.
+        /// <list type="bullet">
+        /// <item>
+        /// If the specified origin has an internationalized domain name (IDN), the punycoded value is used. If the origin
+        /// specifies a default port (e.g. 443 for HTTPS or 80 for HTTP), this will be dropped as part of normalization.
```

Comment:
Except the default port is only dropped for IDNs

Reply:
It says so, doesn't it: 

> If the specified origin has an internationalized domain name (IDN)